content/markdown/03-plotting/03-02-iterative-plotting.rmd

```diff
@@ -0,0 +1,294 @@
+---
+output:
+  pdf_document: default
+  html_document: default
+---
+# Iterative Plotting for Easy and Aesthetic Figures
+**Author:** Cole Brookson
+**Date:** 20 July 2022
+
+```{r, echo=FALSE}
+options(tidyverse.quiet = TRUE)
+```
+
+When making a plot, typically the best way to go about it is to iteratively make a version of the plot, observe the outcome, and then make changes in an iterative fashion. This prevents us from making mistakes in a long set of code that we then can't diagnose the problem with, and further, helps us catch mistakes (typos, colour problems, etc) as we only have to take in a few new components each time, and not a whole pile. 
+
+For our example, let's use the example from the (LINK) Intro to `ggplot2` section, looking at salamanders and trout: 
+
+```{r}
+library(tidyverse)
+library(lterdatasampler)
+
+df <- lterdatasampler::and_vertebrates
+
+names(df)
+```
+
+## Building a Basic Plot
+
+Let's again plot a scatter plot of `length_1_mm` vs. `weight_g`. We'll start out with the absolute minimum number of arguments: 
+
+1. The call to `ggplot()`
+2. The aesthetics (i.e. what variables we want on the x- and y-axis)
+3. The `geom_` function denoting what *type* of plot we're going for
+
+```{r}
+ggplot() +
+  geom_point(data = df, mapping = aes(x = length_1_mm, y = weight_g))
+```
+
+Ok so we see our plot! Perfect. So the first thing to do, before we make any stylistic additions, is to deal with the components of the plot that are *must-haves* to have a legible, complete plot. First step is the axis labels. So to iteratively work up our plot, we'll add that one component on, by using the new function `labs()`:
+
+```{r}
+ggplot() +
+  geom_point(data = df, mapping = aes(x = length_1_mm, y = weight_g)) +
+  labs(x = "Length (mm)", y = "Weight (g)") 
+```
+
+OK good. Now, this is more aesthetic than anything, but we might want to change the appearance of this plot away from this ugly grey background and grid lines. This is a more personal decision regarding what exactly you want your plot to look like. However, there are lots of `themes` available in the `ggplot2` package to choose from. For example, we could use the `theme_classic()` like so: 
```

Comment:
I would say that different fields have different conventions regarding themes. Eg. biology- no horizontal grid lines, no border, no grey background